src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala

```diff
@@ -403,6 +404,19 @@ final case class ProcessApiServiceImpl(
     }
   }
 
+  private def deleteRiskAnalysisOnModeUpdate(newMode: EServiceMode, catalogItem: CatalogItem)(implicit
+    contexts: Seq[(String, String)]
+  ): Future[Unit] = {
+    val shouldRiskAnalyzesBeDeleted: Boolean = catalogItem.mode == Receive && newMode == EServiceMode.DELIVER
+    if (shouldRiskAnalyzesBeDeleted)
+      Future
+        .traverse(catalogItem.riskAnalysis)(risk =>
+          catalogManagementService.deleteRiskAnalysis(catalogItem.id, risk.id)
+        )
+        .map(_ => ())
+    else Future.unit
+  }
```

Comment:
It might be simplified as
```suggestion
  ): Future[Unit] = 
    if (newMode == EServiceMode.DELIVER)
      Future
        .traverse(catalogItem.riskAnalysis)(risk =>
          catalogManagementService.deleteRiskAnalysis(catalogItem.id, risk.id)
        )
        .map(_ => ())
    else Future.unit
 
```

Reply:
This will be applied every time an update is performed without mode change.